src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @author Qishen Zhang
+ */
+
+define(['js/util',
+		'../Libs/EpicEditor/js/epiceditor',
+    	'text!./DocumentEditorDialog.html',
+    	'css!./DocumentEditorDialog.css',
+    	'css!../Libs/EpicEditor/themes/base/epiceditor.css'], 
+	function(Util, 
+			marked,
+			DocumentEditorDialogTemplate){
+
+		var DocumentEditorDialog;
+
+		/**
+		 * DocumentEditorDialog Constructor
+		 * Insert dialog modal into body and initialize editor with
+		 * customized options
+		 */
+		DocumentEditorDialog = function(){
+			// Get Modal Template node for Editor Dialog and append it to body
+			this._dialog = $(DocumentEditorDialogTemplate);
+			this._dialog.appendTo($(document.body));
+
+			// Get element nodes
+	        this._el = this._dialog.find('.modal-body').first();
+	        this._btnSave = this._dialog.find('.btn-save').first();
+	        this._pMeta = this._el.find('#pMeta').first();
+	        this._content = this._pMeta.find('div.controls').first();
+
+	        /* Create Markdown Editor with options, but load() function should be 
+	         * invoked in callback function when container is rendered on DOM */ 
+	        var editorOptions = {
+	        	container: this._content.get(0), // Get raw DOM element
+	        	basePath: 'decorators/DocumentDecorator/Libs/EpicEditor/',
+	        	autogrow: {
+	        		minHeight: 300,
+	        	},
+	        	button: {
+	        		fullscreen: true,
+	        	},
+	        	parser: marked,
+	        };
+	        this.editor = new EpicEditor(editorOptions);
+	        this.text = ''; // Keep track modified text in editor
+		}
+
+		/**
+		 * Initialize DocumentEditorDialog by creating EpicEditor in Bootstrap modal
+		 * window and set event listeners on its subcomponents like save button. Notice
+		 * EpicEditor is created but not loaded yet. The creation and loading of editor 
+		 * are seperated due to the reason decorator component is not appended to DOM within
+		 * its own domain.
+		 * @param  {String} 	text           Text to be rendered in editor initially
+		 * @param  {Function}   saveCallback   Callback function after click save button
+		 * @return {void}              
+		 */
+		DocumentEditorDialog.prototype.initialize = function(text, saveCallback){
```

Comment:
I don't know why the indentation on Github is not consistent with sublime editor. All codes in my sublime editor is well indented with four space. I'll try to figure it out.